### DISCUSSION TITLE
Solving Eigenvalue Problems within Eigenvalue Executioner

### DISCUSSION THREAD

**Discussion by LagrangeW** [idaholab/moose#20239] [Q&A General] [answered]
Respected experts and peers，

I'm solving a simple multi-group neutron diffusion problem within Eigenvalue Executioner, while couldn't get correct eigenvalue by imitating the given example input file. Is there some more detailed input syntax guidance of this executioner?

Any help would be appreciated, thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

This is the best documentation we have on solving eigenvalue problems.
https://mooseframework.inl.gov/source/problems/EigenProblem.html
Input file syntax wise, there is this page: 
https://mooseframework.inl.gov/source/executioners/Eigenvalue.html

Guillaume

**Reply by LagrangeW** [reply]
thanks so much for your reply！

**Comment by AnthonyB08** [comment]
Hi LagrangeW,

Did you have any success in modeling multi-group neutron transport? I too am having difficulty with the eigenvalue problem. Mine seems to not want to converge.

**Reply by LagrangeW** [reply]
Hello，

I‘ve done some simple 2D diffusion benchmarks, and the the calculated value is in good agreement with the reference.

I met some non-converge problems before also, and the everytime it was found to be input error, I give wrong group constants or group cross section.

Maybe you can try to change your input parameters.

**Reply by AnthonyB08** [reply]
When you found your eigenvalue, lambda, how did you go about solving the criticality diffusion problem using MOOSE. All of my simulations come back 0

**Reply by AnthonyB08** [reply]
I understand that the criticality solves for the differential to be equal to zero, but how does one then analyze the transport for the flux? Is this just not a problem of finding criticality for the respected macroscopic cross-section and diffusion coefficient?

**Reply by LagrangeW** [reply]
I use eigenvalue executioner to solve this type of eigenvalue problem, as @[GiudGiud](https://github.com/GiudGiud) suggests, which could solve out eigenvalue together with it's eigenvector, or neutron flux.

**Reply by AnthonyB08** [reply]
One last question Largrange W, what kernel did you use to couple to two groups together? This is my governing problem 

![Capture](https://user-images.githubusercontent.com/55299169/166062480-4a7889e0-b8ff-437b-bd67-a2301286d663.PNG)

**Reply by LagrangeW** [reply]
I think diffusion kernel [https://mooseframework.inl.gov/source/kernels/Diffusion.html](url) can be used in neutron diffusion term(leakage term), and reaction term [https://mooseframework.inl.gov/source/kernels/Reaction.html](url) can be used in other terms (removal, fission and scattering).
